src/core/qgsfontutils.cpp

```diff
@@ -605,3 +605,28 @@ QStringList QgsFontUtils::recentFontFamilies()
   const QgsSettings settings;
   return settings.value( QStringLiteral( "fonts/recent" ) ).toStringList();
 }
+
+void QgsFontUtils::setFontFamily( QFont &font, const QString &family )
+{
+  font.setFamily( family );
+  if ( !font.exactMatch() )
+  {
+    // some Qt versions struggle with fonts with certain unusual characters
+    // in their names, eg "ESRI Oil, Gas, & Water". Calling "setFamilies"
+    // can workaround these issues... (in some cases!)
+    font.setFamilies( { family } );
```

Comment:
@nyalldawson , what a weird workaround for a weird issue :)

Reply:
@nirvn it relates to https://bugreports.qt.io/browse/QTBUG-46322, the fix was to introduce setFamilies in https://codereview.qt-project.org/c/qt/qtbase/+/240070

Reply:
@nyalldawson , k.